src/services/pallets/PalletsForeignAssetsService.ts

```diff
@@ -56,10 +57,23 @@ export class PalletsForeignAssetsService extends AbstractService {
 			if (foreignAssetData) {
 				// remove any commas from multilocation key values e.g. Parachain: 2,125 -> Parachain: 2125
 				const foreignAssetMultiLocationStr = JSON.stringify(foreignAssetData[0]).replace(/(\d),/g, '$1');
-				const foreignAssetMultiLocation = api.registry.createType(
-					'XcmV3MultiLocation',
-					JSON.parse(foreignAssetMultiLocationStr),
-				);
+				const XcmV3MultiLocation = ['StagingXcmV3MultiLocation', 'XcmV3MultiLocation'];
+				let foreignAssetMultiLocation;
+				try {
+					foreignAssetMultiLocation = api.registry.createType(
+						XcmV3MultiLocation[0],
+						JSON.parse(foreignAssetMultiLocationStr),
+					);
+				} catch (e) {
+					try {
+						foreignAssetMultiLocation = api.registry.createType(
+							XcmV3MultiLocation[1],
+							JSON.parse(foreignAssetMultiLocationStr),
+						);
+					} catch (e) {
+						throw new InternalServerError(`Failed to create MultiLocation type for Foreign Asset`);
+					}
+				}
 
 				const assetMetadata = await api.query.foreignAssets.metadata<AssetMetadata>(foreignAssetMultiLocation);
```

Comment:
Is it possible to just put the multilocation in the query as an Object or string and let the API take care of the type creation.

Reply:
Just confirmed it can.